gcloud-java-datastore/src/main/java/com/google/gcloud/datastore/testing/LocalGcdHelper.java

```diff
@@ -205,16 +212,51 @@ void terminate() throws InterruptedException, IOException {
     @Override
     public void run() {
       try {
-        while (reader.readLine() != null) {
-          // consume line
+        boolean readerDone = false;
+        boolean errorReaderDone = false;
+        String currentLog = null;
+        while (!readerDone || !errorReaderDone) {
+          if (!readerDone && reader.ready()) {
+            readerDone = reader.readLine() == null;
+          }
+          if (!errorReaderDone && errorReader.ready()) {
+            String errorOutput = errorReader.readLine();
+            if (errorOutput == null) {
+              errorReaderDone = true;
+            } else {
+              currentLog = processLog(errorOutput, currentLog);
+            }
+          }
         }
       } catch (IOException e) {
         // ignore
       }
     }
 
-    public static ProcessStreamReader start(Process process, String blockUntil) throws IOException {
-      ProcessStreamReader thread = new ProcessStreamReader(process, blockUntil);
+    private static boolean isNewGcdLog(String line) {
+      return line.contains("com.google.apphosting.client.serviceapp.BaseApiServlet")
+          && !line.trim().startsWith("at ");
+    }
+
+    private static String processLog(String currentLine, String currentLog) {
+      if (isNewGcdLog(currentLine)) {
+        if (currentLog != null) {
+          log.info(currentLog.trim());
+        }
+        return "GCD ";
+      } else if (currentLine.startsWith("SEVERE: ")) {
+        return null; // Don't show duplicate error messages from gcd.sh
```

Comment:
If this was not fixed we should have an issue to change the code once a new gcd.sh with the fix is pushed.